src/hooks/useTimeTravel.jsx

```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react'
+
+export default function useTimeTravel(initalVal='2021-01-01') {
+  const [history, setHistory] = useState([initalVal])
+  const [index, setIndex] = useState(0)
+
+  function save(val) {
+    setHistory(prevHistory => {
+      let newHistory = new Array(history.length + 1)
```

Comment:
it wold probably be wise to use `prevHistory` in your `new Array(` just to make 💯 sure that you have access to the most up-to-date state